crates/bitcoin/src/address.rs

```diff
@@ -13,8 +14,10 @@ use secp256k1::{constants::PUBLIC_KEY_SIZE, Error as Secp256k1Error, PublicKey a
 /// A Bitcoin address is a serialized identifier that represents the destination for a payment.
 /// Address prefixes are used to indicate the network as well as the format. Since the Parachain
 /// follows SPV assumptions we do not need to know which network a payment is included in.
-#[derive(Encode, Decode, Clone, Ord, PartialOrd, PartialEq, Eq, Debug, Copy, TypeInfo, MaxEncodedLen)]
-#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize, std::hash::Hash))]
+#[derive(
```

Comment:
Why do you unconditionally implement the serde traits? I realize that it's now possible, but we don't need it in the runtime so imo better not to bloat the wasm blob

Reply:
by adding a feature gate to `Address` type, `interbtc_primitive` types need to be feature-gated for serde, which would indeed require a feature gate for `vault-registry` types. but then `frame_support` requires `Serialize`, & `Deserialize` and throws the error 
```

  error[E0277]: the trait bound `interbtc_primitives::CurrencyId: Serialize` is not satisfied
      --> /Users/nakul/Desktop/Interlay_Work/node/interbtc/crates/vault-registry/src/lib.rs:69:1
       |
  69   | #[frame_support::pallet]
       | ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Serialize` is not implemented for `interbtc_primitives::CurrencyId`
  ...
  648  |         pub minimum_collateral_vault: Vec<(CurrencyId<T>, BalanceOf<T>)>,
       |         ---------------------------------------------------------------- required by a bound introduced by this call
       |
       = help: the following other types implement trait `Serialize`:
                 &'a T
                 &'a mut T
                 ()
                 (T0, T1)
                 (T0, T1, T2)
                 (T0, T1, T2, T3)
                 (T0, T1, T2, T3, T4)
                 (T0, T1, T2, T3, T4, T5)
               and 205 others
       = note: required for `(interbtc_primitives::CurrencyId, <T as currency::Config>::Balance)` to implement `Serialize`
       = note: 1 redundant requirement hidden
       = note: required for `frame_benchmarking::Vec<(interbtc_primitives::CurrencyId, <T as currency::Config>::Balance)>` to implement `Serialize`
  note: required by a bound in `frame_support::serde::ser::SerializeStruct::serialize_field`
      --> /Users/nakul/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.186/src/ser/mod.rs:1871:12
       |
  1871 |         T: Serialize;
       |            ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`


```

Reply:
Why can't we leave the serialization as-is? Currently some types (notably, those that are part of the genesisconfig), but in many other places they are std-only

Reply:
after a bit more digging found out, vault-registry's `GenesisConfig` use `interbtc-primitives` therefore they need to have  Serialize, & Deserialize traits implemented, 
but `vault-registry` depends on `interbtc-primitives` which further depends upon `bitcoin` crate. So [following feature flag](https://github.com/paritytech/substrate/pull/13027/files#diff-9315604b21af73f4d96659db77ab7ccae3b0c90e5bd6e76c442de3696b6ee41cR37-R42) needs to be added in `bitcoin` and `interbtc-primitives` crate 

Changes in `interbtc-primitives` required
- Add the mentioned  [serde feature](https://github.com/paritytech/substrate/pull/13027/files#diff-9315604b21af73f4d96659db77ab7ccae3b0c90e5bd6e76c442de3696b6ee41cR37-R42)  
- All `Serialize`, & `Deserialize` traits should be unable when `std` or `serde` feature is on. 

Changes in `bitcoin` required
- Add the mentioned [serde feature](https://github.com/paritytech/substrate/pull/13027/files#diff-9315604b21af73f4d96659db77ab7ccae3b0c90e5bd6e76c442de3696b6ee41cR37-R42) 
- All `Serialize`, & `Deserialize` traits should be unable when `std` or `serde` feature is on. 

Changes in `vault-registry` required
- unable the feature for primitives
`primitives = { package = "interbtc-primitives", path = "../../primitives", default-features = false, features = ["serde"] } `

I don't know if this is the solution u intend me to build, as it still adds the traits in `runtime`, regarding wasm blob do u want me to check the size increase if the traits are implemented by default and then move ahead?